src/Microsoft.AspNetCore.Server.Kestrel/Http/ListenerPrimary.cs

```diff
@@ -101,12 +112,59 @@ protected override void DispatchConnection(UvStreamHandle socket)
             }
         }
 
+        private void DetachFromIOCP(UvHandle handle)
+        {
+            if (!_tryDetachFromIOCP)
+            {
+                return;
+            }
+
+            // https://msdn.microsoft.com/en-us/library/windows/hardware/ff728840(v=vs.85).aspx
+            const int FileReplaceCompletionInformation = 61;
+            // https://msdn.microsoft.com/en-us/library/cc704588.aspx
+            const uint STATUS_INVALID_INFO_CLASS = 0xC0000003;
+
+            var statusBlock = new IO_STATUS_BLOCK();
+            var socket = IntPtr.Zero;
+            Thread.Loop.Libuv.uv_fileno(handle, ref socket);
+
+            if (NtSetInformationFile(socket, out statusBlock, _fileCompletionInfoPtr,
+                (uint)Marshal.SizeOf<FILE_COMPLETION_INFORMATION>(), FileReplaceCompletionInformation) == STATUS_INVALID_INFO_CLASS)
```

Comment:
Out of curiosity, what's returned for pre-8.1 versions of windows and linux? I'd kinda expect a `DllNotFoundException` on linux.